### PR TITLE
XIVY-11798 implement base of all known browser

### PR DESCRIPTION
**Pull request by ivy-edp**
Placeholder made for all browsers
- Added several browsers (Func, Datatype, Api, CatPathChooser, SQl-Operation, Table Column)
- Widgets for Script-/Macroinputs/areas & ScriptCell adapted so that browsers-properties can be set in the part
- All elements worked through and correct browsers deposited at corresponding fields
- Output-Part with property "additionalBrowsers" added

ToDo:
- For some reason (still unexplained to me) three IT of the RestPart don't run anymore (currently skipped with .skip)
- End-Page: New widget must be created, which provides a simple input field with the browser function
- Signal code: Combox with browser function necessary
- Rest - Restbody: Combox with browser function necessary

### CODE REVIEW

File: packages/editor/src/components/widgets/code-editor/ScriptInput.tsx [hunk 1]
```diff
@@ -12,8 +12,9 @@ const ScriptInput = ({
   editorOptions,
   keyActions,
   modifyAction,
+  browsers,
   ...props
-}: CodeEditorInputProps & { type: string }) => {
+}: CodeEditorInputProps & { type: string; browsers: BrowserType[] }) => {
```
Comment (ivy-lli) [inline packages/editor/src/components/widgets/code-editor/ScriptInput.tsx:17]: I think you can add the browsers attribute to the CodeEditorInputProps type right? Then you don't have to to this on the macro and the script input.

**Review by ivy-lli — COMMENTED** [review]
Looks good 👍 
Some small comments

File: packages/editor/src/components/parts/common/customfield/CustomFieldPart.tsx [hunk 2]
```diff
@@ -1,17 +1,19 @@
 import { CustomFieldConfigType, WfCustomField } from '@axonivy/inscription-protocol';
 import CustomFieldTable from './CustomFieldTable';
 import { PathCollapsible } from '../path/PathCollapsible';
+import { BrowserType } from '../../../../components/browser';
 
 type CustomFieldPartProps = {
   customFields: WfCustomField[];
   updateCustomFields: (customFields: WfCustomField[]) => void;
   type: CustomFieldConfigType;
+  browsers: BrowserType[];
```
[the remark this answers is not in the document]
Comment (ivy-edp) [reply packages/editor/src/components/parts/common/customfield/CustomFieldPart.tsx:10]: yes, you are right. it is the same everywhere

File: packages/editor/src/components/parts/common/parameter/ParameterTable.tsx [hunk 3]
```diff
@@ -22,7 +32,7 @@ const ParameterTable = ({ data, onChange, hideDesc }: ParameterTableProps) => {
       {
         accessorKey: 'type',
         header: header => <SortableHeader header={header} name='Type' />,
-        cell: cell => <EditableCell cell={cell} />
+        cell: cell => <ScriptCell cell={cell} type={cell.row.original.type} browsers={['datatyp']} />
```
[the remark this answers is not in the document]
Comment (ivy-edp) [reply packages/editor/src/components/parts/common/parameter/ParameterTable.tsx:35]: Ah okay. But in this case, it still needs to be invented. i put the whole thing back on the origin-cell and would do the whole thing accordingly in the next PR

File: packages/editor/src/components/browser/DataTypBrowser.tsx [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import { useState } from 'react';
 import { Input } from '../widgets';
 import { UseBrowserImplReturnValue } from './useBrowser';
-export const DATATYPE_BROWSER_ID = 'datatyp' as const;
+export const DATATYPE_BROWSER_ID = 'datatype' as const;
 
 export const useDataTypBrowser = (): UseBrowserImplReturnValue => {
```
Comment (ivy-lli) [inline packages/editor/src/components/browser/DataTypBrowser.tsx:6]: useDataTyp**e**Browser

File: packages/editor/src/components/browser/Browser.tsx [hunk 5]
```diff
@@ -31,18 +30,8 @@ const Browser = ({ open, onOpenChange, types, accept, location }: BrowserProps)
...
-    attrBrowser,
-    cmsBrowser,
-    funcBrowser,
-    dataTypBrowser,
-    catPathChooserBrowser,
-    tableColBrowser,
-    sqlOpBrowser,
-    apiBrowser
-  ];
+  const allBrowsers = [attrBrowser, cmsBrowser, funcBrowser, dataTypBrowser, catPathChooserBrowser, tableColBrowser, sqlOpBrowser];
```
Comment (ivy-lli) [inline packages/editor/src/components/browser/Browser.tsx:34]: dataTyp**e**Browser

**Review by ivy-lli — APPROVED** [review]

**Pull request merged**